### PR TITLE
Handle gpg signing better

### PR DESCRIPTION
**Pull request by DenWav**
Instead of checking whether it was set previously, setting it to false, then setting it back to true if it was true before, just use the command-line argument in git to override the config for that command. Using a variable makes it pretty painless to do.

I have only tested on Linux, and would like others that are able to test on Windows. I won't be able to test macOS until tomorrow.

- [x] Arch Linux - bash 4.4
- [x] Windows 10 1803 - git-for-windows 2.17.0.windows.1 bash 4.4
- [x] Ubuntu 16.04 - bash 4.3
- [x] macOS 10.13 - bash 3.5 / zsh 5.5

### CODE REVIEW

**Comment by aikar** [comment]
do our scripts even work outside git bash?

could we just alias git='git -c commit.gpgsign=false' instead?

**Comment by DenWav** [comment]
That really is essentially what this is doing, just slightly more explicit. I think the variable replacement is more portable though.

**Comment by zachbr** [comment]
❔ = didn't test
✔️ = built
🔐 = tested signing as expected

Arch Linux - bash 4.4 ✔️ 
Windows 10 1803 - git-for-windows 2.17.0.windows.1 bash 4.4 ✔️🔐
Ubuntu 16.04 - bash 4.3 ❔ 
macOS 10.13 - bash 3.5 / zsh 5.5 ✔️🔐

File: scripts/applyPatches.sh [hunk 1]
```diff
@@ -4,8 +4,8 @@
 PS1="$"
 basedir="$(cd "$1" && pwd -P)"
 workdir="$basedir/work"
-gpgsign="$(git config commit.gpgsign || echo "false")"
-applycmd="git am --3way --ignore-whitespace"
+applycmd="$gitcmd am --3way --ignore-whitespace"
+gitcmd="git -c commit.gpgsign=false"
```
Comment (zachbr) [inline scripts/applyPatches.sh]: Looks like this needs to be above applycmd in order for macOS and Travis-CI to build.

**Review by zachbr — APPROVED** [review]

**Pull request merged**